usaspending_api/reporting/v2/views/submission_history.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework.response import Response
+
+from usaspending_api.agency.v2.views.agency_base import AgencyBase, PaginationMixin
+
+from usaspending_api.common.exceptions import UnprocessableEntityException
+from usaspending_api.common.helpers.generic_helper import get_pagination_metadata
+from usaspending_api.submissions.models import SubmissionAttributes
+
+
+class SubmissionHistory(AgencyBase, PaginationMixin):
+    """Returns submission history of the specified agency for the specified fiscal year and period"""
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/reporting/agencies/agency_code/fiscal_year/fiscal_period/submission_history.md"
+
+    @staticmethod
+    def validate_fiscal_period(fiscal_period):
+        fiscal_period = int(fiscal_period)
+        if fiscal_period < 2 or fiscal_period > 12:
+            raise UnprocessableEntityException(f"fiscal_period must be in the range 2-12")
+
+    def get(self, request, toptier_code, fiscal_year, fiscal_period):
+        self.fiscal_year = int(fiscal_year)
+        self.sortable_columns = [
+            "publication_date",
+            "certified_date",
+        ]
+        self.default_sort_column = "publication_date"
+        self.validate_fiscal_period(fiscal_period)
+        results = (
+            SubmissionAttributes.objects.filter(
+                toptier_code=toptier_code, reporting_fiscal_year=fiscal_year, reporting_fiscal_period=fiscal_period,
+            )
+            .order_by("-published_date")
```

Comment:
Is the sorting configurable? Seems hardcoded

Reply:
There's no sorting specified in the API Contract, so I just hardcoded it to ensure that it had an order.

Reply:
I get that. Seems odd to include all of the sorting logic above and not use it. I don't think it would hurt to allow sorting. At the very least there should be a comment explaining why the columns are defined above but not used here